components/style/matching.rs

```diff
@@ -186,6 +186,9 @@ trait PrivateMatchMethods: TElement {
         use context::DISPLAY_CHANGED_FROM_NONE_FOR_SMIL;
         use properties::longhands::display::computed_value as display;
 
+        debug_assert!(restyle_hints.intersects(RESTYLE_SMIL),
```

Comment:
nit: I think it'd be more elegant to just check the `intersects` here and early return, fwiw.